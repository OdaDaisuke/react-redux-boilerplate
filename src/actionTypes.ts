export const ON_APP_INIT = 'ON_APP_INIT'
export const INC_COUNT = 'INC_COUNT'
export const DEC_COUNT = 'DEC_COUNT'