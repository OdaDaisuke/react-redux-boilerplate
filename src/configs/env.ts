export const env = {
  apiBaseUri: process.env.API_BASE_URI as string,
}
