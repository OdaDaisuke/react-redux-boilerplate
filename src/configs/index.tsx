export * from './styles'
export * from './initial_state'
export * from './env'
