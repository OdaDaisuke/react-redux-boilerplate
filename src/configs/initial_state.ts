export interface InitialStateInterface {
  appState: string
  count: number
}

export const initialState: InitialStateInterface = {
  appState: "",
  count: 0,
}
