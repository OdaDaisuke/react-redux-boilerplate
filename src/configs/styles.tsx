export const styles = {
  bg_default: '#ffffff',

  characterStyle: {
    '-webkit-font-smoothing': 'antialiased'
  },

  p: {
    color: '#4baefc',
    fontSize: '1.85rem',
    letterSpacing: '2px',
  },

  button: {
    backgroundImage: 'linear-gradient(90deg, #2196F3, #51b1fd)',
    borderWidth: 0,
    borderRadius: '2px',
    color: '#fff',
    cursor: 'pointer',
    fontSize: '1.05rem',
    letterSpacing: '2px',
    marginTop: '10px',
    marginRight: '10px',
    marginBottom: '10px',
    marginLeft: '10px',
    outline: 'none',
    paddingTop: '12px',
    paddingRight: '40px',
    paddingBottom: '16px',
    paddingLeft: '40px',
    transition: 'all 0.2s',
    ':hover': {
        transform: 'scale(1.03)',
    },
    ':focus': {
      backgroundImage: 'linear-gradient(afeff, #3fafFf, #6fcfff)',
    },
  }
}

