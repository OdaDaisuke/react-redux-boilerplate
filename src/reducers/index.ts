import { combineReducers } from 'redux'
import mainReducer from './main'

const reducers = combineReducers({
  mainReducer,
})

export default reducers
