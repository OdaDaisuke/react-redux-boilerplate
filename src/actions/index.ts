import * as types from '../actionTypes'
import { SampleService } from '../domain'

export const onAppInit = (callback: any, sampleService: SampleService) => {
  sampleService.get(callback)

  return {
    type: types.ON_APP_INIT
  }
}
