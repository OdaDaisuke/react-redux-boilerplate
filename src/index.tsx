import * as React from 'react'
import * as ReactDOM from 'react-dom'
import { Provider } from 'react-redux'
import 'react-hot-loader/patch'
import createBrowserHistory from 'history/createBrowserHistory'
import { hot } from 'react-hot-loader'
import { css, StyleSheet } from 'aphrodite'
import * as configs from './configs'
import { AppRoute } from './route'
import store from './stores'

export class AppContainer {
  initialize(): void {
    const history = createBrowserHistory()
    ReactDOM.render(
      <div className={css(this.styles.body)}>
	<Provider store={store}>
	  <AppRoute history={history} />
	</Provider>
      </div>,
      document.querySelector('#app-container'),
    )
  }

  get styles() {
    return StyleSheet.create({
        body: {
            backgroundColor: configs.styles.bg_default,
            fontFamily: 'Noto Sans Japanese',
            margin: 0,
            minHeight: '100vh',
            padding: 0,
        },
        '*': {
            boxSizing: 'border-box',
        },
        'p, span': {
            '-webkit-font-smoothing': 'antialiased',
        },
    })
  }
}

new AppContainer().initialize()
