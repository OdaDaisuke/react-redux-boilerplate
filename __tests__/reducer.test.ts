import { IAction } from '../src/reducers'
import mainReducer from '../src/reducers/main'
import * as types from '../src/actionTypes'
import { initialState, InitialStateInterface } from '../src/configs'

describe("Reducer", () => {
    test("Pass the ON_APP_INIT", () => {
        const action: IAction = { type: types.ON_APP_INIT }
        const state: InitialStateInterface = mainReducer(initialState, action)
        expect(state.appState).toBe("Ready.")
    })
})