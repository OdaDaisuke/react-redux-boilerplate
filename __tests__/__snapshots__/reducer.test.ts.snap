// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DEC_COUNT 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`INC_COUNT 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;

exports[`ON_APP_INIT 1`] = `
Snapshot Diff:
- Before
+ After

  Object {
-   "appState": "",
+   "appState": "Ready.",
    "count": 0,
  }
`;

exports[`handle initial state 1`] = `
Snapshot Diff:
Compared values have no visual difference.
`;
